app/acapy.js

```diff
@@ -0,0 +1,86 @@
+const { ACAPY_API_KEY, ACAPY_ADMIN_SERVICE } = require('./env')
+
+const querySubWallets = async (subwalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallets`)
+  const search = new URLSearchParams({ wallet_name: subwalletName })
+  url.search = search.toString()
+
+  const queryResponse = await fetch(url, {
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+    },
+  })
+
+  const wallets = await queryResponse.json()
+
+  return wallets.results
+}
+
+const createSubWallet = async (body) => {
+  const fetch = (await import('node-fetch')).default
+
+  const url = new URL(`${ACAPY_ADMIN_SERVICE}/multitenancy/wallet`)
+  const createResponse = await fetch(url, {
+    method: 'POST',
+    headers: {
+      'x-api-key': ACAPY_API_KEY,
+      'content-type': 'application/json',
+    },
+    body: JSON.stringify(body),
+  })
+
+  return createResponse
+}
+
+const getSubWalletToken = async (subWalletName) => {
+  const fetch = (await import('node-fetch')).default
+
+  const wallets = await querySubWallets(subWalletName)
+  if (wallets.length !== 1) {
+    return null
+  }
+
+  const wallet = wallets[0]
```

Comment:
is it always going to be first wallet `primary` one and returned by this endpoint `/multitenancy/wallets`?

Reply:
If you look at the conditional a couple of lines up we've asserted there is only one wallet at this stage. So yes